botocore/operation.py

```diff
@@ -86,8 +113,25 @@ def call(self, endpoint, **kwargs):
         self.session.emit(event, endpoint=endpoint,
                           model=self.model,
                           params=request_dict,
-                          operation=self)
+                          operation=self,
+                          request_signer=signer)
+
+        # This is kind of ugly, but it lets us hook into the request
+        # created event on a per-operation basis without mucking with
+        # the Endpoint/EndpointCreator objects.
+        real_emit = event_emitter.emit
+        def fake_emit(name, **kwargs):
+            if name.startswith('request-created'):
```

Comment:
I am still confused why we are unable to hook into ``request-created`` from operation objects. Could you elaborate? I guess I am not seeing why you cannot just register to the ``request-created`` event. Is it because operation objects share event_emitters across sessions when clients have copies of the event emitter.